### PR TITLE
Don't call change_privilege twice. fix #1757

### LINKED ISSUE #1757: Startup failure after upgrading
- fluentd version 0.14.23
- Ubuntu 14.04

After upgrading from 0.14.15 to 0.14.23, fluentd will no longer start up.  In the logs, I see the following error continuously:
```
2017-11-20 19:19:00 +0000 [error]: #0 unexpected error error_class=Errno::EPERM error="Operation not permitted"
  2017-11-20 19:19:00 +0000 [error]: #0 /var/lib/gems/2.2.0/gems/serverengine-2.0.5/lib/serverengine/privilege.rb:50:in `groups='
  2017-11-20 19:19:00 +0000 [error]: #0 /var/lib/gems/2.2.0/gems/serverengine-2.0.5/lib/serverengine/privilege.rb:50:in `change'
  2017-11-20 19:19:00 +0000 [error]: #0 /var/lib/gems/2.2.0/gems/fluentd-0.14.15/lib/fluent/supervisor.rb:714:in `change_privilege'
  2017-11-20 19:19:00 +0000 [error]: #0 /var/lib/gems/2.2.0/gems/fluentd-0.14.15/lib/fluent/supervisor.rb:500:in `block in run_worker'
  2017-11-20 19:19:00 +0000 [error]: #0 /var/lib/gems/2.2.0/gems/fluentd-0.14.15/lib/fluent/supervisor.rb:661:in `call'
  2017-11-20 19:19:00 +0000 [error]: #0 /var/lib/gems/2.2.0/gems/fluentd-0.14.15/lib/fluent/supervisor.rb:661:in `main_process'
  2017-11-20 19:19:00 +0000 [error]: #0 /var/lib/gems/2.2.0/gems/fluentd-0.14.15/lib/fluent/supervisor.rb:498:in `run_worker'
  2017-11-20 19:19:00 +0000 [error]: #0 /var/lib/gems/2.2.0/gems/fluentd-0.14.15/lib/fluent/command/fluentd.rb:316:in `<top (required)>'
  2017-11-20 19:19:00 +0000 [error]: #0 /usr/lib/ruby/2.2.0/rubygems/core_ext/kernel_require.rb:54:in `require'
  2017-11-20 19:19:00 +0000 [error]: #0 /usr/lib/ruby/2.2.0/rubygems/core_ext/kernel_require.rb:54:in `require'
  2017-11-20 19:19:00 +0000 [error]: #0 /var/lib/gems/2.2.0/gems/fluentd-0.14.15/bin/fluentd:5:in `<top (required)>'
  2017-11-20 19:19:00 +0000 [error]: #0 /usr/local/bin/fluentd:23:in `load'
  2017-11-20 19:19:00 +0000 [error]: #0 /usr/local/bin/fluentd:23:in `<main>'
2017-11-20 19:19:00 +0000 [info]: Worker 0 finished unexpectedly with status 1
```

### PR DESCRIPTION
**Pull request by repeatedly**

### CODE REVIEW

File: lib/fluent/supervisor.rb [hunk 1]
```diff
@@ -568,7 +568,7 @@ def show_plugin_config
 
     def supervise
       # Make dumpable conf, which is set corresponding_proxies for all elements in all worker sections
-      dry_run
```
Comment (mururu) [inline lib/fluent/supervisor.rb]: Supervisor process may have to change privilege to run `configure`. So it seems better that supervisor call `change_privilege` and worker process doesn't call `change_privilege` here:  https://github.com/fluent/fluentd/blob/5a0e14776b737ade4ca5a002bfbd0287d219dcb3/lib/fluent/supervisor.rb#L520

File: lib/fluent/supervisor.rb [hunk 2]
```diff
@@ -545,10 +545,10 @@ def dry_run_cmd
     end
 
     ## Set Engine's dry_run_mode true to override all target_id of worker sections
-    def dry_run
+    def dry_run(with_change_privilege = true)
       begin
         Fluent::Engine.dry_run_mode = true
-        change_privilege
```
Comment (mururu) [inline lib/fluent/supervisor.rb]: Supervisor process may have to change privilege to run `configure`. So it seems better that supervisor call `change_privilege` and worker process doesn't call `change_privilege` here:  https://github.com/fluent/fluentd/blob/5a0e14776b737ade4ca5a002bfbd0287d219dcb3/lib/fluent/supervisor.rb#L520

Comment (repeatedly) [reply lib/fluent/supervisor.rb]: In this situtaion, worker cann't change privilege when pass `--no-supervisor`, right?

Comment (mururu) [reply lib/fluent/supervisor.rb]: Ah, yes. Then it is good to call `change_privilege` on worker process when `--no-privilege` is specified and to call `change_privilege` on supervisor process when `--no-supervisor` isn't specified.

**Comment by repeatedly** [comment]
@mururu change code. Could you check again?

**Comment by mururu** [comment]
Looks good.

**Pull request merged**